tests/fetching/test_strat.py

```diff
@@ -21,35 +21,65 @@ def project_folder():
         yield p
 
 
-def test_if_test_object_searches_for_test_files_in_folders_it_generates_test_files(
+def test_if_test_object_searches_for_test_files_in_subfolders_it_generates_test_files(
     project_folder,
 ):
-    test_object = PyTestStrategy(False)
+    test_object = PyTestStrategy(
+        pathlib.Path.cwd(), overwrite_tests=False, recurse_into_subdirs=True
+    )
     test_object.apply(project_folder)
-    test_file_paths = cwd.glob("test_*.py")
-    new_test_file_paths = [
-        pathlib.Path(str(file_path).replace(".py", test_object.APPENDED_FILEPATH))
-        for file_path in test_file_paths
-        if not str(file_path).endswith(test_object.APPENDED_FILEPATH)
-    ]
+    test_file_paths = list(
+        filter(
+            lambda s: not s.name.endswith(PyTestStrategy.SUFFIX), cwd.rglob("test_*.py")
+        )
+    )
+    new_test_file_paths = list(
+        map(lambda p: p.parent / f"{p.stem}{PyTestStrategy.SUFFIX}", test_file_paths)
+    )
+
+    print(test_file_paths)
+    print(new_test_file_paths)
+
     assert len(new_test_file_paths) != 0
 
     for new_test_file_path in new_test_file_paths:
-        assert new_test_file_path.exists()
+        assert new_test_file_path.exists(), f"{new_test_file_path} does not exist!"
+
+    for new_test_file_path in new_test_file_paths:
+        new_test_file_path.unlink()
 
 
-def test_if_test_object_searches_for_test_files_in_folders_including_subfolders_it_generates_test_files(
+def test_if_test_object_searches_for_test_files_in_folders_including_subfolders_it_overwrites_test_files(
```

Comment:
Add a reset so that the test files won't be filled with decorators. Before overwriting, store the lines of each line. After overwriting, use the lines to regenerate the original files.